src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs

```diff
@@ -75,87 +74,151 @@ private XmlReader ApplyIgnorables(string file)
 				}
 
-				if (adjusted.Contains("{x:Bind"))
-				{
-					// Apply replacements to avoid having issues with the XAML parser which does not
-					// support quotes in positional markup extensions parameters.
-					// Note that the UWP preprocessor does not need to apply those replacements as the
-					// x:Bind expressions are being removed during the first phase and replaced by "connections".
-					adjusted = Regex.Replace(
+				var replacement = "{0}{1} {2}:Ignorable=\"{3}\"".InvariantCultureFormat(targetLine, mcString, mcName, newIgnoredFlat);
+				adjusted = ReplaceFirst(
 						adjusted,
-						"\"{x:Bind.*?}\"",
-						e => e.Value.Replace('\'', XamlConstants.XBindSubstitute)
+						targetLine,
+						replacement
+					)
+					.TrimEnd("\r\n");
+			}
+
+			// Replace the ignored namespaces with unique urns so that same urn that are placed in Ignored attribute
+			// are ignored independently.
+			foreach (var n in newIgnored)
+			{
+				adjusted = adjusted
+					.Replace(
+						"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, document.DocumentElement.GetNamespaceOfPrefix(n)),
+						"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, Guid.NewGuid())
 					);
+			}
+
+			// Put all the included namespaces in the same default namespace, so that the properties get their
+			// DeclaringType properly set.
+			foreach (var n in _includeXamlNamespaces)
+			{
+				var originalPrefix = document.DocumentElement.GetNamespaceOfPrefix(n);
+
+				if (!originalPrefix.StartsWith("using:"))
+				{
+					adjusted = adjusted
+						.Replace(
+							"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, document.DocumentElement.GetNamespaceOfPrefix(n)),
+							"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, document.DocumentElement.GetNamespaceOfPrefix(""))
+						);
 				}
+			}
 
-				return XmlReader.Create(new StringReader(adjusted));
+			foreach (var includedCond in conditionals.IncludedConditionals)
+			{
+				var valueSplit = includedCond.Value.Split('?');
+				// Strip the conditional part, so the namespace can be parsed correctly by the Xaml reader
+				adjusted = adjusted
+					.Replace(
+						includedCond.OuterXml,
+						"{0}=\"{1}\"".InvariantCultureFormat(includedCond.Name, valueSplit[0])
+					);
 			}
 
-			return XmlReader.Create(file);
+			if (hasxBind)
+			{
+				// Apply replacements to avoid having issues with the XAML parser which does not
+				// support quotes in positional markup extensions parameters.
+				// Note that the UWP preprocessor does not need to apply those replacements as the
+				// x:Bind expressions are being removed during the first phase and replaced by "connections".
+				adjusted = Regex.Replace(
+					adjusted,
+					"\"{x:Bind.*?}\"",
+					e => e.Value.Replace('\'', XamlConstants.XBindSubstitute)
+				);
+			}
+
+			return XmlReader.Create(new StringReader(adjusted));
+		}
+
+		private static string ReplaceFirst(string targetString, string oldValue, string newValue)
+		{
+			var index = targetString.IndexOf(oldValue);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)